lib/travis/build/script/nix.rb

```diff
@@ -71,6 +78,25 @@ def announce
         def script
           sh.cmd 'nix-build'
         end
+
+        private
+
+        def is_number?(str)
+          true if Float(str) rescue false
+        end
+
+        def expand_channel_url(channel, url)
+          if is_number? url
+            case channel.to_s
+            when "nixpkgs"
+              "https://nixos.org/channels/nixos-#{url}"
```

Comment:
I think the channel shouldn't be "expanded", it doesn't need to be for the user to get a nice experience: Users can specify `channel:nixpkgs-unstable` or `channel:nixos-18.09-darwin` (this is new as of Nix 2.0) so they don't need to specify a full URL anyway. This can be added to the Nix on Travis documentation (https://github.com/travis-ci/docs-travis-ci-com/blob/master/user/languages/nix.md).

Beyond this nit, this is great work.

Reply:
Very cool, I was not aware that you could do that.  That will significantly simplify this Pr.

I'll go ahead and put a PR into the docs-travis-ci-com repo after this one lands to reflect the change.

Thanks for the feedback!